### PR TITLE
Dimon 46 spring tests

### PR DESCRIPTION
**Pull request by dimonkoval**

### CODE REVIEW

File: src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.model.Role;
+import mate.academy.model.Role.RoleName;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
```
Comment (overpathz) [inline src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:13]: Consider add test for empty role(null)
or get role by name if it's not existing in db

File: src/test/java/mate/academy/dao/impl/UserDaoImplTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        roleUser.setRoleName(RoleName.USER);
+        roleDao.save(roleUser);
+        user = new User();
+        user.setEmail(USER_EMAIL);
+        user.setPassword(USER_PASSWORD);
+        user.setRoles(Set.of(roleUser));
+    }
+
+    @Test
+    void save_Ok() {
```
Comment (overpathz) [inline src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:44]: I think is a little overhead.
You can just compare users and if user is saved he have non null id.

File: src/test/java/mate/academy/dao/impl/UserDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.Role.RoleName;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoImplTest extends AbstractTest {
```
Comment (overpathz) [inline src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:18]: What if I have to save existing user or save user with null email. I think it should be covered by tests

File: src/main/java/mate/academy/model/User.java [hunk 4]
```diff
@@ -63,4 +64,23 @@ public String toString() {
                 + ", email='" + email + '\''
                 + '}';
     }
+
+    @Override
```
Comment (overpathz) [inline src/main/java/mate/academy/model/User.java:68]: You have incorrect overridden of equals & hashcode methods

https://vladmihalcea.com/the-best-way-to-implement-equals-hashcode-and-tostring-with-jpa-and-hibernate/

https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/

File: src/test/java/mate/academy/security/AuthenticationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    private static final String USER_PASSWORD = "password";
+    private UserService userService;
+    private RoleService roleService;
+    private AuthenticationService authenticationService;
+    private Role userRole;
+    private PasswordEncoder passwordEncoder;
+    private User testUser;
+
+    @BeforeEach
+    void setUp() {
```
Comment (overpathz) [inline src/test/java/mate/academy/security/AuthenticationServiceImplTest.java:32]: Iа objects are not modified or used differently in each test method, it can be considered a good practice to move them to a @BeforeAll method.
(passwordEncoder, authenticationService, etc.)
It's related to other tests as well.
And since these are tests, you can do them final and initialize as mock when declare

File: src/test/java/mate/academy/validation/PasswordValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        Password mockPassword = mock(Password.class);
+        when(mockPassword.field()).thenReturn(PASSWORD_NAME);
+        when(mockPassword.fieldMatch()).thenReturn(REPEAT_PASSWORD_NAME);
+        passwordValidator.initialize(mockPassword);
+        dto = new UserRegistrationDto();
+        dto.setPassword(PASSWORD_VALUE_VALID);
+        dto.setRepeatPassword(PASSWORD_VALUE_VALID);
+        context = mock(ConstraintValidatorContext.class);
+    }
+
```
Comment (overpathz) [inline src/test/java/mate/academy/validation/PasswordValidatorTest.java:34]: How about test that checking that repeated password is not equals?

**Review by overpathz — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]
Good job!
